### PR TITLE
Migrate NCChatViewController to Swift

### PR DESCRIPTION
**Pull request by SystemKeeper**
This PR migrates our `NCChatViewController` to Swift and tries to decouple its responsibilities. 

With this PR we have 3 ViewControllers:

#### `InputbarViewController` 
* Subclass of `SLKTextViewController`
* Responsible for the appearance of the ViewControllers and the TextInputbar.
* Contains everything needed for a TextInputbar and AutoCompletion
* Allows initialization with a UIView instaed of a UITableView (Note: The constraints are not finished yet, but generally it's working)

Snippet to test:
```objc
    NCRoom *lastRoom = [NCRoom allObjects].lastObject;
    NCRoom *lastRoomUnmanaged = [[NCRoom alloc] initWithValue:lastRoom];
    UIView *testView = [[UIView alloc] init];
    [testView setBackgroundColor:[UIColor systemRedColor]];
    InputbarViewController *inputbarVC = [[InputbarViewController alloc] initFor:lastRoomUnmanaged withView:testView];
    NCNavigationController *navigationController = [[NCNavigationController alloc] initWithRootViewController:inputbarVC];
    [self presentViewController:navigationController animated:YES completion:nil];
```

#### `BaseChatViewController`
* Subclass of `InputbarViewController`
* Responsible for everything chat related, that does not need a ChatController, e.g. cell-rendering, cell height calculation, etc.
* Allows initialization with an Array of `NCChatMessage` for displaying arbitrary chat messages (e.g. when selecting a message in a search result and then obtaining the context of that message)

Snippet to test:
```objc
    NCRoom *lastRoom = [NCRoom allObjects].lastObject;
    NCRoom *lastRoomUnmanaged = [[NCRoom alloc] initWithValue:lastRoom];

    NSPredicate *query = [NSPredicate predicateWithFormat:@"accountId = %@ AND token = %@", lastRoomUnmanaged.accountId, lastRoomUnmanaged.token];

    RLMResults *managedMessages = [NCChatMessage objectsWithPredicate:query];
    RLMResults *managedSortedMessages = [managedMessages sortedResultsUsingKeyPath:@"messageId" ascending:YES];
    NSMutableArray *sortedMessages = [NSMutableArray new];

    for (NSInteger i = 0; i < 90; i++) {
        NCChatMessage *sortedMessage = [[NCChatMessage alloc] initWithValue:managedSortedMessages[i]];
        [sortedMessages addObject:sortedMessage];
    }

    BaseChatViewController *controller = [[BaseChatViewController alloc] initFor:lastRoomUnmanaged withMessage:sortedMessages];
    // OR: BaseChatViewController *controller = [[BaseChatViewController alloc] initFor:lastRoomUnmanaged withMessage:sortedMessages withHighlightId:917];
    NCNavigationController *navigationController = [[NCNavigationController alloc] initWithRootViewController:controller];

    [self presentViewController:navigationController animated:YES completion:nil];
```

#### `ChatViewController`
* Subclass of `BaseChatViewController`
* Basically the replacement for our `NCChatViewController`
* Includes everything that's needed to send and receive new chat messages.

**General remarks**:
* The chat cells have been changed to not contain a reference to `NextcloudTalk-Swift.h` from an `.h` file (instead forward declaration and then use in the `.m` file)
* Some files have been changed to include `nullable`/`nonnullable` annotations and type references. Xcode warns that not all of the other properties also contain these annotations now, but this would be out of scope of this PR.

### CODE REVIEW

**Comment by SystemKeeper** [comment]
Fixed:
* VoiceMessage playback (a `continue` was missing when updating the cell)
* Modifying reactions from BaseChatViewController (moved the functionality to ChatViewController)
* Fixed crash when receiving no reactions (side effect of the second item, the server returns an array, but we expect a dictionary)

**Review by Ivansss — APPROVED** [review]
